zephyr/lc3_config.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2022 Bose Corporation All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LIB_LIBLC3CODEC_ZEPHYR_LC3_CONFIG_H_
+#define LIB_LIBLC3CODEC_ZEPHYR_LC3_CONFIG_H_
+
+/* This file is a c-wrapper for the c++ LC3 lib. Please refer to the C++ API documentation
+ * for a full description of the API.
+ * General usage:
+ * 1) Call _create to allocate memory for the c++ object using the c++-lib allocation mechanism.
+ * 2) Use the wrapper functions as needed passing the allocated object as first parameter.
+ * 3) After usage call _destroy to free up the memory.
+ *    Invalid usage of the object reference after a call to _destroy cannot be detected and will
+ *    most likely result in a crash. */
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+#define LC3_EXTERN_C extern "C"
+#else
+#define LC3_EXTERN_C
+#endif
+
+typedef void* Lc3Config_t;
```

Comment:
It is because in c we cannot store c++ objects - hence it is normal to store the object references in void pointers and have a thin c++ wrapper layer that converts the c-calls to c++ calls (into the objects).